server/src/main/java/org/candlepin/controller/ContentManager.java

```diff
@@ -75,45 +116,56 @@ public ContentManager(ContentCurator contentCurator, ProductCurator productCurat
      *  given owner
      *
      * @throws NullPointerException
-     *  if the provided content entity is null
+     *  if entity or owner is null
      *
      * @return
      *  a new Content instance representing the specified content for the given owner
      */
     @Transactional
     public Content createContent(Content entity, Owner owner) {
-        log.debug("Creating new content for org: {}, {}", entity, owner);
-
         if (entity == null) {
-            throw new NullPointerException("entity");
+            throw new IllegalArgumentException("entity is null");
         }
 
-        Content existing = this.contentCurator.lookupById(owner, entity.getId());
+        if (owner == null) {
+            throw new IllegalArgumentException("owner is null");
+        }
+
+        log.debug("Creating new content for org: {}, {}", entity, owner);
+
+        Content existing = this.ownerContentCurator.getContentById(owner, entity.getId());
+
+        // TODO: FIXME:
+        // There's a bug here where if changes are applied to an entity's collections, and then
+        // this method is called, the check below will trigger an illegal state exception, but
+        // Hibernate's helpful nature will have persisted the changes during the lookup above.
+        // This needs to be re-written to use DTOs as the primary source of entity creation, rather
+        // than a bolted-on utility method.
 
         if (existing != null) {
             // If we're doing an exclusive creation, this should be an error condition
             throw new IllegalStateException("Content has already been created");
         }
 
         // Check if we have an alternate version we can use instead.
-        List<Content> alternateVersions = this.contentCurator.getContentByVersion(
-            entity.getId(), entity.hashCode()
-        );
+        List<Content> alternateVersions = this.contentCurator
+            .getContentByVersion(entity.getId(), entity.hashCode())
+            .list();
```

Comment:
Any reason you are returning criteria instead of just returning the results from the curator?